elements/pl-file-editor/pl-file-editor.js

```diff
@@ -43,6 +44,35 @@ window.PLFileEditor = function(uuid, options) {
         this.editor.setOption('maxLines', Infinity);
     }
 
+    const default_preview_text = '<p>Begin typing to preview</p>';
+    if (options.preview == 'markdown') {
+        let preview = this.element.find('.preview')[0];
+        let editor = this.editor;
+        let conv = new showdown.Converter();
+
+        let update_preview = function() {
+            let text = editor.getValue();
+            if (text.trim().length == 0) {
+                preview.innerHTML = default_preview_text;
+            } else {
+                preview.innerHTML = conv.makeHtml(editor.getValue());
+                if (text.includes('$') ||
+                    text.includes('\\(') || text.includes('\\)') ||
+                    text.includes('\\[') || text.includes('\\]')) {
+                    MathJax.typesetPromise();
+                }
+            }
+        };
+        editor.session.on('change', function() {
+            update_preview();
```

Comment:
If you put a huge markdown document in the editor and type some more text in it, then the previewing gets very slow.

One way to get around this is to have two flags: one for `preview_in_progress` and one for `preview_needed`. 

On change, if `preview_in_progress == True` then we set `preview_needed = True` and return. If `preview_in_progress == False` then we set `preview_in_progress = True` and call `update_preview()`.

At the end of `update_preview()` we set `preview_in_progress = False` and also check `preview_needed`. If `preview_needed == True` then we set `preview_in_progress = True, preview_needed = False` and we call `update_preview()` again.